app/controllers/reports/ndbn_annuals_controller.rb

```diff
@@ -6,6 +6,17 @@ def index
 
   def show
     raise ActionController::RoutingError.new('Not Found') unless validate_show_params
+
+    @year = params[:year]
```

Comment:
I realize you're just adding on to existing action code here, but this should _probably_ be moved into strong params. 

Instead of raising a RoutingError, we can do something like: 

```ruby
def year_params
  params.require(:year)
end
```

We don't need to validate that the year is positive -- if the data is bad it just won't find anything, but having it as a required field in strong params means that we can at least expect that it won't raise a 500 because there's *no* data for the param.